contrib/plugins/qt-creator-project.py

```diff
@@ -0,0 +1,231 @@
+          shutil.rmtree(symlinkDir)
+       os.makedirs(symlinkDir)
+
+    # regex for all source / header files
+    source  = re.compile(".*\.[ch](pp)?")
+
+    # lists for storing all found sources files / include directories
+    sList = []
+    hList = []
+
+    # now go through all checkout dirs to find all header / include files
+    for name,path in OrderedDict(sorted(dirs, key=lambda t: t[1])).items():
+        newPath = os.path.join(symlinkDir, name)
+        os.symlink(os.path.join(os.getcwd(), path), newPath)
+        for root, directories, filenames in os.walk(newPath):
+            for filename in filenames:
+                if source.match(filename):
+                    sList.append(os.path.join(os.getcwd(), os.path.join(root,filename)))
+            # it's more faster to add all directories to includes and not to use regex, sort & remove duplicate entries
+            # this also helps qt-creator to resolve includes like <linux/bitops.h> which
+            # is not found in case there is no header in 'linux'
+            hList.append(os.path.join(os.getcwd(),root))
+
+    # get system default project id
+    # storred in ~/.config/QtProject/qtcreator/profiles.xml
+    # <value type="QString" key="PE.Profile.Id">{4bf97dcc-3970-4b8f-97cc-b4ce38f2b2d7}</value>
+    projectid = -1
+    with open(os.path.join(expanduser('~'), ".config/QtProject/qtcreator/profiles.xml")) as profilesFile:
+        for line in profilesFile:
+            if 'PE.Profile.Id' in line:
+                m = re.search('.*PE.Profile.Id">(.+?)</value>', line)
+                if m:
+                    projectid = m.group(1)
+    if projectid == -1:
+       print("No profile file found. Generated projects may not work")
+
+    # Generate the includes file
+    includesFile = os.path.join(destination, projectDir + ".includes")
+    generateFile(hList, includesFile)
+    # Generate files file
+    filesFile = os.path.join(destination, projectDir  + ".files")
+    generateFile(sList, filesFile)
+
+    if not updateOnly:
+        #Generate creator file
+        creatorFile = os.path.join(destination, projectDir  + ".creator")
+        generateFile([], creatorFile)
+        #Generate the creator.shared file using a template and modify some settings
+        sharedFile = os.path.join(destination, projectDir  + ".creator.shared")
+        templateFile = os.path.join(os.getcwd() + "/plugins/qt-project.template")
+        with open(sharedFile, 'w') as sharedFile:
+            with open(templateFile) as template:
+                for line in template:
+                    if 'plugin:id' in line:
+                        sharedFile.write(line.replace("<!-- plugin:id -->", projectid ))
+                    elif 'plugin:buildstep' in line:
+                        addBuildSteps(sharedFile, 'dev' if devMode else 'build', project, destination)
+                    else:
+                        sharedFile.write(line)
+
+def projectGenerator(argv, bobRoot):
```

Comment:
The signature should look like: `generator(package, argv)`.

The parsing of the recipes and the package selection must be done outside of the plugin. They are not part of the stable plugin API. It would be done by all generators anyway. Just spill the extra aguments into `argv` here (see below).

If `bobRoot` will be used in the future is also not clear. Better not pass it.